src/commands/deploy.ts

```diff
@@ -441,14 +440,74 @@ export default class Deploy extends DeployCommand {
       return;
     }
 
-    cli.action.start(chalk.blue('Deploying'));
-    await Promise.all(
-      approved_pipelines.map(async (pipeline) => {
-        await PipelineUtils.pollPipeline(this.app.api, pipeline.pipeline.id);
-        this.log(chalk.green(`${pipeline.component_name} Deployed`));
-      })
-    );
-    cli.action.stop();
+    let component_deployments: Deployment[] = [];
+    for (const approved_pipeline of approved_pipelines) {
+      const { data: pipeline_deployments } = await this.app.api.get(`/pipelines/${approved_pipeline.pipeline.id}/deployments`);
+      component_deployments = component_deployments.concat(pipeline_deployments.filter((d: Deployment) => d.type === 'component'));
+    }
+
+    const environment_graph = (await this.app.api.get(`/environments/${environment.id}/graph`)).data;
+    const service_nodes = environment_graph.nodes.filter((n: DependencyNode) => n.__type === 'service');
+    const deployment_tasks: { [s: string]: ListrTask<any>[] } = {};
+    for (const service_node of service_nodes) {
+      const { component_account_name, component_name, service_name, tag } = ServiceVersionSlugUtils.parse(service_node.config.name);
+      const component_version_name = ComponentVersionSlugUtils.build(component_account_name, component_name, tag);
+
+      if (!deployment_tasks[component_version_name]) {
+        const component_deployment = component_deployments.find(d => d.id === service_node.deployment_id);
+        if (!component_deployment) {
+          continue; // skip, service already exists in environment and isn't currently being deployed
+        }
+        deployment_tasks[component_version_name] = [{
+          title: `Pipeline ${component_deployment.pipeline.id}`,
+          task: () => PipelineUtils.pollPipeline(this.app.api, component_deployment.pipeline.id),
+        }];
+      }
+      deployment_tasks[component_version_name].push({
+        title: `Service ${service_name}`,
+        task: () => new Promise((resolve, reject) => {
+          let service_health_poll_count = 0;
+          const service_health_poll = setInterval(async () => {
+            if (service_health_poll_count > 180) {
+              clearInterval(service_health_poll);
+              return reject(new Error(`Service ${service_name} deployment timed out`));
+            }
+
+            const environment_health: EnvironmentHealth = (await this.app.api.get(`/environments/${environment.id}/health`)).data;
```

Comment:
I'm wondering if we can do this loop on the server so we only have to poll a single endpoint instead of polling for each service. The endpoint could return the health of each service environment?

Reply:
> instead of polling for each service
The endpoint used is to get the health of the environment which includes all services. 

> service environment
service in the environment?